hazelcast/src/main/java/com/hazelcast/internal/config/YamlConfigSchemaValidator.java

```diff
@@ -68,18 +71,52 @@ public void validate(YamlMapping rootNode) {
         try {
             // this could be expressed in the schema as well, but that would make all the schema validation errors much harder
             // to read, so it is better to implement it here as a semantic check
-            long definedRootNodeCount = PERMITTED_ROOT_NODES.stream()
+            List<String> definedRootNodes = PERMITTED_ROOT_NODES.stream()
                     .filter(rootNodeName -> rootNode.child(rootNodeName) != null)
-                    .count();
-            if (definedRootNodeCount != 1) {
+                    .collect(toList());
+            if (definedRootNodes.size() != 1) {
                 throw new SchemaViolationConfigurationException(
                         "exactly one of [hazelcast], [hazelcast-client] and [hazelcast-client-failover] should be present in the"
-                                + " root schema document, " + definedRootNodeCount + " are present",
+                                + " root schema document, " + definedRootNodes.size() + " are present",
                         "#", "#", emptyList());
+            } else {
+                validateAdditionalProperties(rootNode, definedRootNodes.get(0));
             }
             SCHEMA.validate(YamlToJsonConverter.convert(rootNode));
         } catch (ValidationException e) {
             throw wrap(e);
         }
     }
+
+    private void validateAdditionalProperties(YamlMapping rootNode, String hzConfigRootNodeName) {
+        if (!PERMITTED_ROOT_NODES.contains(hzConfigRootNodeName)) {
+            throw new IllegalArgumentException(hzConfigRootNodeName);
+        }
+        ObjectSchema schema = (ObjectSchema) SCHEMA;
+        Set<String> forbiddenRootPropNames = ((ObjectSchema) schema.getPropertySchemas()
+                .get(hzConfigRootNodeName)).getPropertySchemas().keySet();
+        List<String> misIndentedRootProps = new ArrayList<>();
+        rootNode.children().forEach(yamlNode -> {
+            if (forbiddenRootPropNames.contains(yamlNode.nodeName())) {
+                misIndentedRootProps.add(yamlNode.nodeName());
+            }
+        });
+        if (misIndentedRootProps.isEmpty()) {
+            return;
+        }
+        if (misIndentedRootProps.size() == 1) {
+            String propName = misIndentedRootProps.get(0);
+            throw createExceptionForMisIndentedConfigProp(propName);
+        } else {
+            List<SchemaViolationConfigurationException> causes = misIndentedRootProps.stream()
+                    .map(this::createExceptionForMisIndentedConfigProp)
+                    .collect(toList());
+            throw new SchemaViolationConfigurationException(causes.size() + " schema violations found", "#", "#", causes);
```

Comment:
This is a tough question. I agree it would be good to detect alignment issues, but this makes the HZ configuration not embeddable in configs that should have valid `map`, `network` etc nodes at root level. I think this check is safe only if under the given node in `misIndendedRootProps` there is a valid HZ configuration. 
So I would choose from one of these:
1) Skip this indentation check
2) Replace the exception with a warning
3) Perform a schema validation with all `misIndendedRootProps` "candidate" as if it was under the detected HZ config root node type. This requires a bit of DOM manipulation, but I think it's doable. 

I think 3) is the best from UX point of view. If it turned out not to be too difficult, I'd give it a try. In theory, it's still possible to produce false positives in this case, but the chances that someone comes up with a valid HZ config internally is kind of low, and if it happens, the schema validation can be turned off. If we want to be 100% sure, we can take a sort of "meta-section" at the top-level into account, which section lists the root-level nodes this check should ignore. Or we can do this instead of 3), so

```yaml
network:
  ...
hazelcast:
  map:
    ...
# this section helps the Hazelcast config validator to prevent false errors
hazelcast-validation-ignore-roots:
  - network
  - ...
```